org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/staging/AbapGitStagingView.java

```diff
@@ -1192,8 +1217,19 @@ private void handleException(Exception e, final String action) {
 		//invalid credentials : this condition check is only valid from 2002
 		if (e instanceof ResourceException && GitCredentialsService.isAuthenticationIssue((ResourceException) e)) {
 			AbapGitStagingView.this.repositoryCredentials = null;
-			ErrorHandlingService.handleExceptionAndDisplayInErrorDialog(e, getSite().getShell());
-			if (getGitCredentials(e.getLocalizedMessage()).equals(Status.OK_STATUS)) {
+
+			String message = e.getLocalizedMessage();
+			//If credentials not retrieved from secure storage, show appropriate Error Dialog
+			//Otherwise simply show the credentials dialog
+			if (this.credsRetrievedFromSecureStorage == false) {
+				ErrorHandlingService.handleExceptionAndDisplayInErrorDialog(e, getSite().getShell());
+			} else {
+				GitCredentialsService.deleteCredentialsFromSecureStorage(this.repository.getUrl());
+				message = null;
```

Comment:
Again note that for performing any action (compare, stage, push), if the creds were retrieved from secure secure storage, 
we no longer show the Credentials Dialog filled with creds retrieved from secure storage (to avoid security leak).
In case these creds are incorrect, showing an `ErrorDialogWithDetails` right after performing an action, looks weird without the involvement of the Credentials Dialog.
Hence,
1) a flag is used to keep track whether the credentials were retrieved from Secure Storage.(Comments to better handle it are welcome.)
2) Since, the creds are incorrect, creds are deleted from Secure Storage.
3) Instead of Error Dialog, directly a fresh and empty `Credentials Dialog`  is shown to the user to get the correct credentials.